### PR TITLE
Moves testing to setup.sh

### PR DESCRIPTION
**Pull request by anselmbradford**
Decouples `grunt test` from `grunt build` and moves it to `setup.sh`.

The reason for this is:
  - Now that we we're using /dist/, running `grunt build` happens more frequently.
  - Our test suite is growing larger/slower, so having it on-demand becomes more important.
  - Travis is running our tests so will catch errors if `grunt test` is forgotten to run before submitting a PR (which should be done before a PR is opened).

## Changes

- Removes test task from `grunt build` pipeline and moves it to `setup.sh`.
- Moves order of test task below lintjs task in Gruntfile.

## Testing

- `./setup.sh` should run test suites.
- `grunt build` should not.
- `grunt test` should.

## Review

- @jimmynotjim 
- @KimberlyMunoz 
- @sebworks

### CODE REVIEW

**Comment by jimmynotjim** [comment]
If we merge #780, should `setup.sh` just run `test.sh`?

**Comment by KimberlyMunoz** [comment]
I'm concerned that not having a task that runs both building and testing tasks will mean we run tests locally less often. I'd rather have a `grunt dev` task for all the CSS, JS and Copy related tasks.

**Comment by jimmynotjim** [comment]
Thinking about it, I have to agree with Kim, but maybe that just means `grunt build` should build, `grunt test` should test, and the default `grunt` should do both.

**Comment by imuchnik** [comment]
I strongly feel that if you are not running  your tests all the time, it sort of defeats the purpose. I also strongly feel that developers should have a fast and tight feedback loop about the quality of the software they produce. With that  in mind, I think unit tests should remain in the build, so that it just gets run and it is one less thing developers need to remember to do. On the other hand, slower browser tests should be moved and ran regularly and on demand and we should also thrive to make our unit tests small, fast and air tight. 

In conclusion, whatever default task is to run regularly should run tests.

**Comment by anselmbradford** [comment]
I figured this would generate some controversy :) Yeah, the idea is just to give more control over what happens when. There are clearly cases where edits do not need to run all the testing and linting suites, and it's just inefficient to do so.

I browsed through some other projects to see what they did around their building tasks. Owning-a-home has `build`, `release`, `ship`, and `deploy` tasks that are related to building and releasing steps. Eregs uses `test`, `test-js`, `build`, `squish`. Dash uses `build` and `test`, with a default of those two run together (like what Jimmy mentioned). Sentia uses `test`, `testServerSide`, `frontEndTest`, `frontAndServerSideTest`, and `build`, which runs the `frontAndServerSideTest` task. So... the short of it is there's no real cfpb standard that I can see. 

Rails, which I think is good to look to for these kind of things, has by default three environments: "development", "test", and "production". If we took from this model, we have `grunt test` to test, `grunt build` for production, and then what Kim said, we would have `grunt dev` for development (which would probably eventually be good to set up to not do things like minimize the assets either). 

Additionally, we could have the testing more granular in the type of test to run and have default task be full build + test. Thoughts on any of that?

**Comment by jimmynotjim** [comment]
What's the difference between release, ship and deploy in OaH, they all sound the same? That's kind of my issue with a build also running _all_ the tests, because name wise, it sounds like it's just building the project. If everyone else is ok with it though, go with it.

**Comment by anselmbradford** [comment]
> What's the difference between release, ship and deploy in OaH, they all sound the same? That's kind of my issue with a build also running all the tests, because name wise, it sounds like it's just building the project. If everyone else is ok with it though, go with it.

Yeah the OAH setup is confusing to me to in the way you're describing. They are all variations on building, testing, and moving and minifying assets. I'd vote for what you described for `grunt build` in https://github.com/cfpb/cfgov-refresh/pull/779#issuecomment-122957546, the difference to `grunt dev` then would be asset minification. This is mostly to streamline how `grunt watch` runs, which I don't personally use, but I know @jimmynotjim and @KimberlyMunoz use it I think? How would you like to see that working?

**Comment by KimberlyMunoz** [comment]
So long as there's a test, build and both task that we ask folks to run before submitting a PR, I'm good. 

At the risk of re-doing this work, I think we should toss this question on the front-end repo after we've made a decision to see if we can standardize across repos. (Or at least have a standard that new projects will aim towards.)

**Comment by jimmynotjim** [comment]
@KimberlyMunoz I like that idea, we might want send a PR to the generator rather than an issue. `code > talk`

**Comment by KimberlyMunoz** [comment]
Let's at least cross-post to the Front End repo so that other FEWDs who aren't following all the Capital Framework repos can see it and direct them to comment at the generator. There should be a related task for the front-end repo anyway so we remember to add some documentation about it there.

**Comment by jimmynotjim** [comment]
@anselmbradford want to get this one in for this sprint, then I can update the gulp tasks to match the decisions made here?

**Comment by anselmbradford** [comment]
@jimmynotjim I pushed the default update in https://github.com/cfpb/cfgov-refresh/commit/59e8a6266ce0e2b941090c7be056c2bfea1d171d, but upon further reflection it may make more sense to leave out `grunt dev` here because I'd need to separate minification out of the less task, etc., which sounds like it'll get usurped by https://github.com/cfpb/cfgov-refresh/pull/797.

**Comment by jimmynotjim** [comment]
:+1:

**Comment by KimberlyMunoz** [comment]
Sounds good. `grunt dev` was more of a nice to have. I'll make a note to update our documentation to run `grunt` before submitting a PR.

**Comment by KimberlyMunoz** [comment]
:+1:

**Comment by anselmbradford** [comment]
Yeah `dev` would still be good to add, just in the gulp flow.
